### ISSUE TITLE
Completion: substring matching bug?

### ISSUE THREAD

**Issue by alphapapa** [fish-shell/fish-shell#1854]
I just built and installed fish 2.1.1, but the substring fuzzy matching doesn't seem to work properly.

```
/tmp/bpfish$ fish --version
fish, version 2.1.1

/tmp/bpfish$ ls
buildresult/               fish_2.1.1.dfsg-2_i386.changes           fish_2.1.1.dfsg-2~ubuntu14.04.1_source.build    fish-trusty/
fish-2.1.1.dfsg/           fish_2.1.1.dfsg-2_i386.deb               fish_2.1.1.dfsg-2~ubuntu14.04.1_source.changes
fish_2.1.1.dfsg-2.diff.gz  fish_2.1.1.dfsg-2~ubuntu14.04.1.diff.gz  fish_2.1.1.dfsg.orig.tar.gz
fish_2.1.1.dfsg-2.dsc      fish_2.1.1.dfsg-2~ubuntu14.04.1.dsc      fish-dbg_2.1.1.dfsg-2_i386.deb

/tmp/bpfish$ ls deb<tab>
/tmp/bpfish$ ls fish
```
According to #568, the substring matching should cause the two deb packages to be the autocomplete suggestions, but instead the first suggestion is `fish`.

**Comment by ridiculousfish** [comment]
Did this get better with 701a37fd00a37563bd4645e6a0c93144dd1b0551?

**Comment by ridiculousfish** [comment]
I think this was addressed by 701a37fd00a37563bd4645e6a0c93144dd1b0551. Closing. Thanks for reporting this.
